transcom/mymove
# INT B-20608 Update CAC validation page with approved text and sign out button

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20608)

> [!NOTE]  
> You will need an Okta account in order to thoroughly test this functionality. If your account is locked, hit me up and I'll break the chains.

## Summary

When a customer is required to authenticate with a smart card on initial registration, they are redirected to an informational page that only shows if:
1. The feature flag is turned on
2. The service member has `false` in the `cac_validated` column in the `service_members` table
3. They're logged in
The logic is found in `/src/scenes/MyMove/index.jsx:183`
```
            {/* Showing Smart Card info page until user signs in with SC one time */}
            {userIsLoggedIn && !cacValidated && cacValidatedFeatureFlag && <SmartCardRedirect />}
```

Previously the page told the user to sign out and didn't offer any tools to save some time. They simply signed out of MilMove, redirected back to the MilMove sign in page, then signed in again, accepted terms, etc.

This backlog item does the following:
- adds a new endpoint `/auth/logoutOktaRedirect` with logic in `auth.go` called `NewLogoutOktaRedirectHandler` <br /> This handler sends the user back to `/auth/okta` instead of `/sign-in`, which is unique for the added `Sign Out` button also in this PR
- refactors the `SmartCardRedirect` component to implement navigation as well as redirecting the user to the Okta logout endpoint with the return value sent back from `NewLogoutOktaRedirectHandler`
- adds the new endpoint for the UI to access
- adds/updates all relevant tests

### How to test

1. In your `envrc`, flip `FEATURE_FLAG_CAC_VALIDATED_LOGIN` to `true`
2. Slap in `direnv allow` to tell your server what is up
3. Strike a match and fire up `make server_run` and `make client_run`
4. Take a ride into the customer application and DON'T use local sign in
5. Sign in as a bonafide Okta user, create an account, get the code, etc.
6. You'll be redirected back to MilMove and you'll see the new & hot `SmartCardRedirect` component
7. Review the Agility ticket to make sure all information needed is in the blurbs
8. Click `Sign Out` (the one in the middle of the page) and go on an adventure
9. Verify that you're redirected to the Okta sign in page and NOT MilMove's - we don't wanna be there. It's scary.

## Screenshots
![Screenshot 2024-07-18 at 9 01 26 AM](https://github.com/user-attachments/assets/e2da5cf3-31c5-435c-bc02-a0be5eef774d)


https://github.com/user-attachments/assets/fde4b982-e0ab-4218-85b2-d0535e588ef3
